sdk/java/src/main/java/com/chain/http/Client.java

```diff
@@ -141,13 +143,15 @@ public Client(URL url, String accessToken) {
    * @return the result of the post request
    * @throws ChainException
    */
-  public <T> BatchResponse<T> batchRequest(String action, Object body, Type tClass, Type eClass)
-      throws ChainException {
-    return post(
-        action,
-        body,
-        (Response response, Gson deserializer) ->
-            new BatchResponse(response, deserializer, tClass, eClass));
+  public <T> BatchResponse<T> batchRequest(
+      String action, Object body, final Type tClass, final Type eClass) throws ChainException {
+    ResponseCreator<T> rc =
+        new ResponseCreator<T>() {
+          public T create(Response response, Gson deserializer) throws ChainException, IOException {
+            return (T) new BatchResponse<T>(response, deserializer, tClass, eClass);
```

Comment:
Does this actually work at runtime? There appear to be two competing meanings of the generic specifier `T` in this method.

1. The type of the object returned by the ResponseCreator lambda class. In this case, it should be `ResponseCreator<BatchResponse<T>>`
2. The type of the object contained in the `BatchResponse` generic class.

My assumption would be that the code in the PR will break at runtime, but this might work:

```
ResponseCreator<BatchResponse<T>> rc =
        new ResponseCreator<BatchResponse<T>>() {
          public BatchResponse<T> create(Response response, Gson deserializer) throws ChainException, IOException {
            return new BatchResponse<T>(response, deserializer, tClass, eClass);
          }
        }
```